### PR TITLE
[WIP] Complete Entity removal

### PR DESCRIPTION
**Pull request by mcdonnelldean**
- removed mem-store as a default loaded plugin,
- removed all tests specifically relating to entitys,
- corrected tests that used entities to also load mem-store
- removed `lib/store.js` as it's to be moved to seneca-entity

**DO NOT MERGE** - This is awaiting a change from @mihaidma to add the golden circle as a test range in travis so we can assess the actual impact.

### CODE REVIEW

**Comment by mcdonnelldean** [comment]
@rjrodger Can you through a quick eye over?

store.js has been moved to entity with a back compat check to ensure we don't break entity `0.0.1` or seneca versions pre this change.

This won't be merged until after mihai's test changes since they test the plugins. Once in and if your happy I will merge.

**Comment by mcdonnelldean** [comment]
@geek Uhoh, looks like we have an nsp check. Could you check it? I've no access to it 💃

File: test/seneca.test.js [hunk 1]
```diff
@@ -1170,6 +1170,7 @@ describe('seneca', function () {
     si.options({errhandler: done})
 
     si.use('entity')
+    si.use('mem-store')
```
Comment (mihaidma) [inline test/seneca.test.js]: @mcdonnelldean don't forget about the mem store enabled by default. You are right that would be a kind of (small) breaking change

Comment (mcdonnelldean) [inline test/seneca.test.js]: Working on it now. I need to release another version of entity before I can remove that. I wanted to rebase your tests in first. Once I up the version of entity I can drop that reference and run it against some of the actual stores to verify it's all good 👍  This won't be going in until fully checked.

**Comment by mcdonnelldean** [comment]
Waiting on https://github.com/senecajs/seneca-entity/pull/21 and a change to line 1173

**Comment by mcdonnelldean** [comment]
This is breaking Seneca Transport on Node 5 only. Need to investigate

**Comment by mcdonnelldean** [comment]
Rerunning these as this change doesn't break on #434 and that worries me greatly.

**Comment by mcdonnelldean** [comment]
Now passing.

**Pull request merged**